### PR TITLE
fix: update dependency jest when to v3

### PR DESCRIPTION
**Pull request by JeffreyMa597**
What problem does this PR solve?
------
As title.

Ref: https://github.com/Magickbase/neuron/pull/162

Check List
------
1、List
2、Detail

### Test
e2e Test
### Task
none

### CODE REVIEW

File: packages/neuron-wallet/tests/block-sync-renderer/indexer-cache-service.intg.test.ts [hunk 1]
```diff
@@ -10,6 +10,17 @@ import RpcService from '../../src/services/rpc-service'
 const stubbedGetTransactionFn = jest.fn()
 const stubbedGetHeaderFn = jest.fn()
 const stubbedGetTransactionsByLockScriptFn = jest.fn()
+const stubbedCollectFn = jest.fn(() => {
```
Comment (yanguoyu) [inline packages/neuron-wallet/tests/block-sync-renderer/indexer-cache-service.intg.test.ts:13]: I think this is redundant. It has mock at https://github.com/nervosnetwork/neuron/pull/2684/files#diff-200ae59642cd5bd8413700f7b04c2ca44c1e965c58cc03b7a11cd8765b3740e9R147-R161

File: packages/neuron-wallet/tests/block-sync-renderer/indexer-cache-service.intg.test.ts [hunk 2]
```diff
@@ -88,7 +101,7 @@ const mockGetTransactionHashes = (mocks: any[] = []) => {
 
   for (const lock of [formattedDefaultLockScript, formattedAcpLockScript, formattedLegacyAcpLockScript]) {
     const { hashes } = mocks.find(mock => mock.lock === lock) || { hashes: [] }
-    stubbedConstructor.calledWith(expect.anything(), { lock }).mockReturnValue({
+    stubbedConstructor.calledWith(expect.anything(), { lock }, undefined, { includeStatus: false }).mockReturnValue({
```
Comment (yanguoyu) [inline packages/neuron-wallet/tests/block-sync-renderer/indexer-cache-service.intg.test.ts:104]: It's better to mock the `this.indexer.ckbRpcUrl` as a string, but not undefined.

File: packages/neuron-ui/.eslintrc.js [hunk 3]
```diff
@@ -96,5 +97,6 @@ module.exports = {
         allowExpressions: true,
       },
     ],
+    "no-multi-spaces": "error",
```
Comment (WhiteMinds) [inline packages/neuron-ui/.eslintrc.js:100]: This is unnecessary as code formatting is handled by Prettier, so ESLint should focus on handling non-style-related issues or complex style problems that Prettier cannot handle.

**Review by WhiteMinds — CHANGES_REQUESTED** [review]

File: packages/neuron-ui/.eslintrc.js (hunk 3 above)
Comment (JeffreyMa597) [reply packages/neuron-ui/.eslintrc.js:100]: > This is unnecessary as code formatting is handled by Prettier, so ESLint should focus on handling non-style-related issues or complex style problems that Prettier cannot handle.

I'll move it to the config of `prettier`. Thx.

File: packages/neuron-wallet/src/block-sync-renderer/sync/indexer-cache-service.ts [hunk 4]
```diff
@@ -1,3 +1,4 @@
+/* eslint-disable no-console */
```
Comment (WhiteMinds) [inline packages/neuron-wallet/src/block-sync-renderer/sync/indexer-cache-service.ts:1]: Need to remove.

File: packages/neuron-wallet/tests/block-sync-renderer/indexer-cache-service.intg.test.ts [hunk 5]
```diff
@@ -108,6 +123,8 @@ const fakeTx3 = {
   txStatus: { status: 'committed', blockHash: fakeBlock2.hash },
 }
 
+
+
```
Comment (WhiteMinds) [inline packages/neuron-wallet/tests/block-sync-renderer/indexer-cache-service.intg.test.ts:127]: Need to remove.

File: packages/neuron-wallet/tests/block-sync-renderer/indexer-cache-service.intg.test.ts [hunk 6]
```diff
@@ -88,7 +103,7 @@ const mockGetTransactionHashes = (mocks: any[] = []) => {
 
   for (const lock of [formattedDefaultLockScript, formattedAcpLockScript, formattedLegacyAcpLockScript]) {
     const { hashes } = mocks.find(mock => mock.lock === lock) || { hashes: [] }
-    stubbedConstructor.calledWith(expect.anything(), { lock }).mockReturnValue({
+    stubbedConstructor.calledWith(expect.anything(), { lock }, ckbRpcUrl, { includeStatus: false }).mockReturnValue({
```
Comment (WhiteMinds) [inline packages/neuron-wallet/tests/block-sync-renderer/indexer-cache-service.intg.test.ts:106]: I see that the type definition of `TransactionCollector` is as follows. Are these four parameters reasonable?
```
declare class TransactionCollector {
  constructor(
    indexer: Indexer,
    queries: QueryOptions,
    options?: TransactionCollectorOptions
  );
}
```

**Review by WhiteMinds — CHANGES_REQUESTED** [review]

File: packages/neuron-ui/.eslintrc.js [hunk 7]
```diff
@@ -41,6 +41,7 @@ module.exports = {
         argsIgnorePattern: '^_',
       },
     ],
+
```
Comment (yanguoyu) [inline packages/neuron-ui/.eslintrc.js:44]: You can remove this change.

**Review by yanguoyu — APPROVED** [review]

File: packages/neuron-wallet/tests/block-sync-renderer/indexer-cache-service.intg.test.ts (hunk 6 above)
Comment (JeffreyMa597) [reply packages/neuron-wallet/tests/block-sync-renderer/indexer-cache-service.intg.test.ts:106]: It's from the '@nervina-labs/ckb-indexer'.  Maybe you find the base definition?

image
image

Comment (WhiteMinds) [reply packages/neuron-wallet/tests/block-sync-renderer/indexer-cache-service.intg.test.ts:106]: Oh, indeed.

**Review by WhiteMinds — APPROVED** [review]

**Comment by Keith-CY** [comment]
Conflicted, and please pay attention to the advice at https://github.com/nervosnetwork/neuron/pull/2684#discussion_r1211222077

File: packages/neuron-wallet/tests/block-sync-renderer/indexer-cache-service.intg.test.ts [hunk 8]
```diff
@@ -18,10 +30,13 @@ const stubbedRPCServiceConstructor = jest.fn().mockImplementation(() => ({
 
 const stubbedIndexerConstructor = jest.fn().mockImplementation(() => ({
   getTransactionsByLockScript: stubbedGetTransactionsByLockScriptFn,
```
[the remark this answers is not in the document]
Comment (JeffreyMa597) [reply packages/neuron-wallet/tests/block-sync-renderer/indexer-cache-service.intg.test.ts:32]: LGTM, updated. Thx

**Comment by JeffreyMa597** [comment]
> Conflicted, and please pay attention to the advice at [#2684 (comment)](https://github.com/nervosnetwork/neuron/pull/2684#discussion_r1211222077)

Updated, Thx

**Review by devchenyan — APPROVED** [review]

**Pull request merged**